Source/CarthageKit/Project.swift

```diff
@@ -811,9 +831,14 @@ public final class Project { // swiftlint:disable:this type_body_length
 			}
 
 		return loadResolvedCartfile()
-			.map { resolvedCartfile -> [(Dependency, PinnedVersion)] in
+			.flatMap(.latest) { resolvedCartfile in
+				return self.transitiveDependencies(dependenciesToCheckout, resolvedCartfile: resolvedCartfile)
+			}
+			.zip(with: loadResolvedCartfile())
```

Comment:
This loads the resolved cartfile again. Can you reuse `resolvedCartfile` inside the `flatMap` instead?

```swift
.flatMap(.latest) { resolvedCartfile in
    return self
        .transitiveDependencies(dependenciesToCheckout, resolvedCartfile: resolvedCartfile)
        .map { ($0, resolvedCartfile) }
}
```

That'll save reading the cartfile from disk again.

Reply:
This makes sense. It didn't feel right to load the Cartfile again, but I didn't know a better way.